Mobile/ShopLens/Droid/Activities/CameraActivity.cs

```diff
@@ -187,127 +176,113 @@ protected override void OnActivityResult(int requestCode, Result resultCode, Int
+                        return new RecognitionResult { Predictions = await classifyImageTask };
+                    }
                     
-                }
-            })
+                    var recognizeTextTask = GetTextFromImageAsync(image);
+
+                    await Task.WhenAll(classifyImageTask, recognizeTextTask);
+
+                    var ocrResult = recognizeTextTask.Result;
+                    var weightString = new RegexMetricWeightSubstringFinder().FindWeightSpecifier(ocrResult);
+
+                    var recognitionResult = new RecognitionResult
+                    {
+                        RawOcrResult = ocrResult,
+                        Predictions = classifyImageTask.Result,
+                        WeightSpecifier = weightString
+                    };
+                    return recognitionResult;
+                })
                 .ContinueWith(task =>
                 {
                     progressBar.Visibility = ViewStates.Gone;
                     if (task.IsFaulted)
                     {
                         System.Diagnostics.Debug.WriteLine(task.Exception);
+                        return;
                     }
-                    if (task.IsCompletedSuccessfully)
-                    {
-                        shoppingCartErrorDialog.Show();                        
-                    }
+
+                    var recognitionResult = task.Result;
+                    var thingsToSay = new List<string>
+                        {$"This is {recognitionResult.BestPrediction}", recognitionResult.WeightSpecifier};
+                    
+                    tts.Speak(
+                        string.Join(". ", thingsToSay),
+                        QueueMode.Flush,
+                        null,
+                        null);
+
+                    var ocrResult = recognitionResult.RawOcrResult;
+                    if (!string.IsNullOrEmpty(ocrResult))
+                        new MessageBarCreator(rootView, $"OCR: {ocrResult}").Show();
+
+                    GetShoppingCartAddItemDialog(recognitionResult).Show();
                 }, TaskScheduler.FromCurrentSynchronizationContext());
         }
 
-        public void AddToShoppingCart()
+        private void AddToShoppingCart(string guess)
         {
-            prefs.AddString(guess.First().ToString().ToUpper() + guess.Substring(1));
-            shoppingCartMessageBar.Show();
+            var preferences = new ActivityPreferences(this, prefsName);
+            preferences.AddString(guess.FirstCharToUpper());
+            shoppingCartItemAddedMessageBar.Show();
         }
 
         // When the current voice recognition session stops.
         public void OnResults(Bundle results)
         {
             var matches = results.GetStringArrayList(SpeechRecognizer.ResultsRecognition);
-            if (matches.Count > 0)
+            if (matches.Count <= 0) return;
```

Comment:
I always write the brackets with the if statement, even when its one line. I guess this writing style is better.

Reply:
Some reviewers always say it's better to write braces, others say it's better to omit braces. Probably should be noted in the coding standard.